_targets.R

```diff
@@ -27,5 +38,27 @@ list(
               filepath_local = water_volume_logpile_csv,
               filepath_log = 'out/water_volume_logpile_log.csv'),
     format = "file"
+  ),
+  tar_target(
+    # images to upload to s3
+    water_images,
+    image_data %>%
+      distinct(Feature, image_file, image_credit) %>%
+      filter(!is.na(image_file))
+  ),
+  tar_target(
+    image_upload_log, {
+      file_name <- water_images$image_file
+      file_s3 <- sprintf('visualizations/images/%s', file_name)
+      file_local <- sprintf('Images/%s', file_name)
+      file_parse <- gsub('.', '_', file_name, fixed = TRUE)
+      file_log <- sprintf("out/image_%s_log.csv", file_parse)
+      s3_upload(filepath_s3 = file_s3,
+                on_exists = "replace",
+                filepath_local = file_local,
+                filepath_log = file_log)
```

Comment:
Rather than a different log file per image (since there are so many), I think it would be cleaner to do each upload and have `s3_upload()` return the tibble of `local_filepath`, `s3_filepath`, and `timestamp`. Then, make a separate target that writes a single CSV called `out/image_log.csv`. Would require a few small changes to what you currently have but would mean uploading only one CSV to GitHub:

1. Change `s3_upload()` to return the tibble by deleting the `filepath_log` argument and the last two lines that call `write_csv()` and `return()`.
2. Adjust `water_volume_logpile_log_csv` and `image_upload_log` target to not pass in an argument for `filepath_log`. Also, delete `format = "file"` from these `tar_target()` calls.
3. Decide if you want only one "upload log" file containing both images and the water volume logpile info.
4. (if "yes" to 3) Make a new target called `upload_log` and have it `bind_rows()` for the target that uploaded the pools and fluxes CSV & the target that uploaded all the images then writes a single CSV file. (if "no" to 3) Make two new targets that take the image and pools/fluxes CSV log tibbles and individually writes those tibbles to two different files. Don't forget to include `format = "file"` on these new targets!

This is a totally optional step, but simplifies the `out/` part of this repo!

Reply:
yeah much better like that